File: src/BookStore.Domain/Book.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BookStore.Domain
{
    public class Book : Entity
    {
        public string  Name { get; set; }
        public string  Author { get; set; }
        public string  Description { get; set; }
        public double Value { get; set; }
        public DateTime PublishedDate { get; set; }
        public int CategoryId { get; set; }
        /* EF Relation */
        public Category Category { get; set; }
    }
}

File: BookStore-SPA/src/app/_models/Book.ts
export class Book{
    
        id: number;
        name: string;
        author: string;
        description: string;
        value: number;
        publishDate: any;
        categoryid: number;
        constructor(
            id: number,
              name: string, 
              author: string,
              description: string,
              value: number,
              publishDate: any,
              categoryid: number) {
                this.id = id;
                this.name = name;
                this.author = author;
                this.description = description;
                this.value = value;
                this.publishDate = publishDate;
                this.categoryid = categoryid;
            }
    
    }